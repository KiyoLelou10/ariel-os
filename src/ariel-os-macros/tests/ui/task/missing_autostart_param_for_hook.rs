#![no_main]
#![feature(impl_trait_in_assoc_type)]

// FAIL: using hooks require the task to be autostart
#[ariel_os::task(usb_builder_hook)]
async fn main() {}
