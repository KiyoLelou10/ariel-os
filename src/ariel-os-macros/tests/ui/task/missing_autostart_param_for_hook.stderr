error: custom attribute panicked
 --> tests/ui/task/missing_autostart_param_for_hook.rs:5:1
  |
5 | #[ariel_os::task(usb_builder_hook)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: the task must be `autostart` to instantiate hooks
